Fakebook.Posts/Fakebook.Posts.DataAccess/FakebookPostsContext.cs

```diff
@@ -103,14 +103,14 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                         Id = 1,
                         UserEmail = "john.werner@revature.net",
                         Content = "Just made my first post!",
-                        CreatedAt = DateTimeOffset.Now
+                        CreatedAt = DateTime.Now
```

Comment:
why?

Reply:
If i didnt change it, our test fails because SQLLITE doesn't handle datetimeoffset. Any suggestions on how i can fix that issue?

Reply:
search around, there are solutions for the use case.
@nosyrbllewe we do need to be handling dates/times better. as is, i'm pretty sure time zone info is getting obliterated and changed across several layers in the app. solutions include - using DateTimeOffset throughout, which isn't currently being done here, and - using DateTime in the EF model but DateTimeOffset everywhere else, and being careful to convert to UTC on the way in, and mark as UTC on the way out.

this concrete error is what would need to be addressed if we went with datetimeoffset in the ef model.

Reply:
The way I envisioned it working is to have the database only store the UTC times, but then convert it to a local time based upon a user's time zone setting on the way out. I personally feel it is a flaw to have the database store anytime zone information on the timestamps as it will likely create issues or extra complications to sync times between two users located in different time zones. Though there are some small drawbacks with storing time without the time zone as mentioned [here](https://stackoverflow.com/questions/11537106/is-it-always-a-good-idea-to-store-time-in-utc-or-is-this-the-case-where-storing), but I don't think it would cause any issue for us.

Reply:
a DateTime does not uniquely identify a moment in time, unless we manually ensure that any time we handle in that type is UTC. a DateTimeOffset always uniquely identifies a moment in time. both data types map to the same PostgreSQL type by default: "timestamp with time zone". that data type, and DateTimeOffset, and the JSON string made by System.Text.Json, and the JS Date object, all of these correctly process time zones automatically. the big danger with DateTime is forgetting that when you deserialize anything with time zone data into that type, anywhere, the time zone info is simply lost and you need to offset it if you know it wasn't UTC. if the client sends a date that doesn't use the user's time zone setting, which we aren't storing anywhere right now, we'd interpret it wrongly. that consideration applies to the REST interface, not so much the ORM interface, where as long as we consistently convert to UTC we'll be fine.

all that said i'll leave the choices here to you

Reply:
That makes sense. We should probably stick with DateTimeOffset then. Looking around, I found a work-a-round for DateTimeOffset for SQLite [here](https://blog.dangl.me/archive/handling-datetimeoffset-in-sqlite-with-entity-framework-core/).  This should be able to fix our issues while also maintaining DateTimeOffset.

Reply:
i changed it back , and we found a fix.